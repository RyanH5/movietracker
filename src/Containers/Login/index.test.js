import {Login} from './index';
import { shallow, mount } from 'enzyme';
import React from 'react';

describe('Login', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  });
  
  
  describe('handleChange', () => {
    let spy;
    let wrapper;

    beforeEach(() => {
      wrapper = mount(<Login />);
      spy = jest.spyOn(wrapper.instance(), 'handleChange');
    });

    it('should update state of email', () => {
      const mockEvent = {
        target: {
          value: 'h',
          name: 'email'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      wrapper.find('input').first().simulate('change'); 

      expect(spy).toHaveBeenCalled();
    });

    it('should update state of password', () => {
      const mockEvent = {
        target: {
          value: 'h',
          name: 'password'
        }
      };
      wrapper.instance().handleChange(mockEvent);
      wrapper.find('input').last().simulate('change'); 

      expect(spy).toHaveBeenCalled();
    });



  });
});