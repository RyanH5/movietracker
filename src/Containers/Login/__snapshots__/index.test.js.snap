// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h2>
          LOGIN
</h2>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="text"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="text"
          value=""
/>,
        <button
          className="login-btn"
>
          Login
</button>,
      ],
      "className": "login-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LOGIN",
        },
        "ref": null,
        "rendered": "LOGIN",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "login-btn",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h2>
            LOGIN
</h2>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="text"
            value=""
/>,
          <button
            className="login-btn"
>
            Login
</button>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-btn",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
